File: 01-Iterations/binaryGap.js
// https://app.codility.com/programmers/lessons/1-iterations/binary_gap/
// results - https://app.codility.com/demo/results/training2FAAGH-ADU/

// you can write to stdout for debugging purposes, e.g.
// console.log('this is a debug message');

function solution(N) {
  // write your code in JavaScript (Node.js 8.9.4)
  let b = N.toString(2);
  let clear = false
  let count = 0;
  let maxCount = 0;
  for (let digit of b) {
      if (digit === '0') {
          count += 1;
      } else {
          if (clear !== false) {
              maxCount = Math.max(maxCount, count);
          } else {
              clear = true
          }
          count = 0;
      }
  }
  return maxCount;
}

File: 01-Iterations/binaryGap.py
import sys

def main():
  n = sys.argv[1]
  return solution(int(n))

def solution(N):
  b = bin(N)[2:]
  print(b)
  clear = False
  count = 0
  maxCount = 0
  for c in b:
    if (c == '1' and count != 0 and clear != False):
      if (count > maxCount):
        maxCount = count
      count = 0
    if (c == '1' and count == 0):
      clear = True
    else:
      count += 1
  print(maxCount)
  return maxCount

main()